featuretools/utils/recommend_primitives.py

```diff
@@ -0,0 +1,242 @@
+        for time series problems. Because there are so few primitives that are only applicable for time series, we recommend users try all of them.
+
+    Note:
+        This function currently only works for single table and will only recommend transform primitives.
+    """
+    es_dataframe_list = list(entityset.dataframe_dict)
+    if len(es_dataframe_list) == 0:
+        raise IndexError("No DataFrame in EntitySet found. Please add a DataFrame.")
+    if len(es_dataframe_list) > 1:
+        raise IndexError(
+            "Multi-table EntitySets are currently not supported. Please only use a single table EntitySet.",
+        )
+
+    target_dataframe_name = es_dataframe_list[0]
+
+    recommended_primitives = set()
+
+    if not include_time_series_primitives:
+        excluded_primitives += TIME_SERIES_PRIMITIVES
+
+    all_trans_primitives = get_transform_primitives()
+    selected_trans_primitives = [
+        p for name, p in all_trans_primitives.items() if name not in excluded_primitives
+    ]
+
+    valid_primitive_names = [
+        prim.name
+        for prim in get_valid_primitives(
+            entityset,
+            target_dataframe_name,
+            1,
+            selected_trans_primitives,
+        )[1]
+    ]
+
+    recommended_primitives.update(
+        _recommend_non_numeric_primitives(
+            entityset,
+            target_dataframe_name,
+            valid_primitive_names,
+        ),
+    )
+    skew_numeric_primitives = set(["square_root", "natural_logarithm"])
+    valid_skew_primtives = skew_numeric_primitives.intersection(valid_primitive_names)
+
+    valid_time_series_primitives = set(TIME_SERIES_PRIMITIVES).intersection(
+        valid_primitive_names,
+    )
+
+    numeric_tags_only_df = entityset[target_dataframe_name].ww.select("numeric")
+
+    if valid_skew_primtives:
+        recommended_primitives.update(
+            _recommend_skew_numeric_primitives(
+                numeric_tags_only_df,
+                valid_skew_primtives,
+            ),
+        )
+
+    if valid_time_series_primitives:
+        recommended_primitives.update(valid_time_series_primitives)
+    return list(recommended_primitives)
+
+
+def _recommend_non_numeric_primitives(
+    entityset: EntitySet,
+    target_dataframe_name: str,
+    valid_primitives: List[str],
+) -> set:
+    """Get a set of non-numeric primitives for a given dataset and a list of primitives.
+
+    Description:
+        Given a single table entity set with a `target_dataframe_name` and an applicable list of `valid_primitives`,
+        get a set of primitives which produce non-unique features.
+
+    Args:
+        entityset (EntitySet): EntitySet that only contains one dataframe.
+        target_dataframe_name (str): Name of target dataframe to access in `entityset`.
+        valid_primitives (List[str]): List of primitives to calculate and check output features.
+    """
+
+    recommended_non_numeric_primitives = set()
```

Comment:
I really like that the functions are typed. Could we add a type annotation here to make it clear this set will hold strings? I think it could aid comprehension.